### PR TITLE
Master

### PR DESCRIPTION
**Pull request by sotechn**

### CODE REVIEW

**Comment by chrisyue** [comment]
hello.

thanks for this PR.

this project follow the [gitflow's](https://github.com/nvie/gitflow) rules so this PR should merge into develop, not master.

and, the coding standard / style should pass the [StyleCi](https://styleci.io/)'s check

**Comment by sotechn** [comment]
Hello.
you can independently make my changes?

**Comment by chrisyue** [comment]
sure, I am creating a new PR to apply your improvement.

File: Parser.php [hunk 1]
```diff
@@ -86,8 +88,14 @@ private function content2Data($content)
...
             }
 
-            if (preg_match('/^#EXTINF:(.+),/', $line, $matches)) {
+            if (preg_match('/^#EXTINF:(\d+),(.+)/', $line, $matches)) {
                 $data['playlist'][$mediaSequence]['duration'] = +$matches[1];
+                $data['playlist'][$mediaSequence]['name'] = $matches[2];
+                continue;
+            }
+
+            if (preg_match('/^#EXTGRP:(.+)/', $line, $matches)) {
```
Comment (chrisyue) [inline Parser.php:97]: could you please tell me what EXTGRP tag is about?

I can't find anything about it on the [HLS draft](https://tools.ietf.org/html/draft-pantos-http-live-streaming-20)

File: Parser.php [hunk 2]
```diff
@@ -51,12 +51,14 @@ public function parse($content)
...
                 $mediaSequence + $index,
+				$row['name'],
+				$row['group'],
                 !empty($row['isDiscontinuity'])
             );
             $playlist->add($mediaSegment);
         }
 
-        return new M3u8($playlist, $version, $targetDuration);
+        return new M3u8($playlist, $version, 0);
```
Comment (chrisyue) [inline Parser.php:61]: why replace `$targetDuration` with `0`?

File: Parser.php (hunk 1 above)
Comment (sotechn) [reply Parser.php:97]: This is group of the channel: movie, music, history, and other. Many IPTV vendors support this tag.

Comment (chrisyue) [reply Parser.php:97]: are there documentations about this tag ??

Comment (chrisyue) [reply Parser.php:97]: I will open a new pr if the documentation of this tag can be found

**Pull request closed without merging**